src/background/controller/wallet.ts

```
@@ -231,10 +231,37 @@ export class WalletController extends BaseController {
     await this.createKeyringWithMnemonics(password, mnemonic);
     // We're creating the Flow address for the account
     // Only after this, do we have a valid wallet with a Flow address
-    await openapiService.createFlowAddress();
+    const result = await openapiService.createFlowAddressV2();
+    this.checkForNewAddress(result.data);
+  };
+
+  checkForNewAddress = async (txid: string): Promise<FclAccount | null> => {
+    // Usage example with a transaction result
+    const txResult = await fcl.tx(txid).onceSealed();
+
+    // Find the AccountCreated event in the events array
+    const accountCreatedEvent = txResult.events.find(
+      (event) => event.type === 'flow.AccountCreated'
+    );
+
+    const extractAddressFromAccountCreatedEvent = (event) => {
+      if (event.type === 'flow.AccountCreated') {
+        return event.data.address;
+      }
+      return null;
```

You should throw an error here